src/hooks/ChatStore.ts

```diff
@@ -0,0 +1,48 @@
+import { CustomerThread } from "@/components/Chat/queries";
+import { Message } from "@/types/supabase";
+import { create } from "zustand";
+import { immer } from "zustand/middleware/immer";
+
+// Define the type for the chat state
+type ChatState = {
+  activeThread: string | null;
+  threads: Record<CustomerThread["id"], CustomerThread>;
+  messages: Record<Message["thread_id"], Record<Message["id"], Message>>;
+};
+
+type ChatActions = {
+  setActiveThread: (threadId: string) => void;
+  updateThreads: (threads: CustomerThread[]) => void;
+  updateMessages: (messages: Message[]) => void;
+};
+
+export const useChatStore = create<ChatState & ChatActions>()(
+  immer((set) => ({
+    activeThread: null,
+    threads: {},
+    messages: {},
+    setActiveThread: (threadId) => {
```

Comment:
- `activeThread` should be renamed to `activeThreadId`
- should also add a `getActiveThread` action since accessing via `state.threads[activeThreadId]` is somewhat clunky